.ado/azure-pipelines.yml

```diff
@@ -87,24 +85,14 @@ jobs:
 
       - task: NodeTool@0
         inputs:
-          versionSpec: "10.x"
+          versionSpec: "12.x"
```

Comment:
Should the NodeTool task be added to the setup template? If you're going to do it every time, seems like a good idea. Alternatively, I think there's a way to request agents that already have some set of software installed, might be worth looking into.

Reply:
Agents already come with a set of software installed, see [this link ](https://github.com/actions/virtual-environments/blob/master/images/win/Windows2019-Readme.md)for windows-2019. With this, I wanted to specify we wanted the 12. version.

Also, I thought about adding it to the "setup-repo" template, but I thought it falls more under a "setup-environment" template, since it isn't dealing with our repo directly. I couldn't think of any other command to add to the "setup-environment" template so I left it out. Thoughts? 

I am open to moving it to "setup-repo" if you think that's the better route.

Reply:
Yeah I forgot that demands are only applicable to self-hosted agents. I don't think it matters too much whether it's a part of a setup-repo or setup-environment (it's harder to make this a generally useful template). The argument could be made that it's a pre-req to having a functioning repository and should be in the setup-repo template.

Again I don't think it really matters where it goes, just that if it's something every job needs it should go into one of the templates.

Reply:
I added the Node+NuGet installs to setup-repo template in the newest commit.